src/Microsoft.Health.Fhir.Core/Features/Search/SearchOptionsFactory.cs

```diff
@@ -151,6 +161,29 @@ public SearchOptions Create(string resourceType, IReadOnlyList<Tuple<string, str
 
             options.UnsupportedSearchParams = unsupportedSearchParameters;
 
+            options.CompartmentType = null;
+
+            // Initialize enum to undefined value
+            var parsedCompartmentType = (CompartmentType)(-1);
+
+            if (!string.IsNullOrWhiteSpace(compartmentType) &&
+    !Enum.TryParse(compartmentType, out parsedCompartmentType))
+            {
+                throw new NotSupportedException(string.Format(Core.Resources.CompartmentTypeIsInvalid, compartmentType));
+            }
+
+            if (Enum.IsDefined(typeof(CompartmentType), parsedCompartmentType))
```

Comment:
You don't really need this either since if it was successfully parsed by TryParse above, prasedCompartmentType will have a valid value here.